app/gui/src/project-view/components/TooltipTrigger.vue

```diff
@@ -1,16 +1,23 @@
 <script setup lang="ts">
-import { useTooltipRegistry, type TooltipDisplayStrategy } from '@/providers/tooltipRegistry'
+import { useTooltipRegistry } from '@/providers/tooltipRegistry'
 import { usePropagateScopesToAllRoots } from '@/util/patching'
-import { Placement } from '@floating-ui/vue'
+import type { Placement } from '@floating-ui/vue'
+import type { VueInstance } from '@vueuse/core'
 import { toRef } from 'vue'
+import type { Opt } from '@/util/data/opt'
 
 const {
   placement = 'top',
-  when = 'always',
+  whenOverflow = undefined,
   showOnClick = false,
 } = defineProps<{
   placement?: Placement
-  when?: TooltipDisplayStrategy
+  /**
+   * If set, the tooltip is inhibited unless the reference element is overflowing. If `true`, the
+   * element in the default slot is the reference element; if an element is explicitly provided, it
+   * will be used as the reference element.
+   */
+  whenOverflow?: Opt<true | HTMLElement | VueInstance>
```

Comment:
From usage perspective, I think this prop could be called `whenOverflowOnly` and take `Opt<boolean | ...>` so false is also valid (maybe even without `Opt`)

Or

make it quite explicit with union of objects with discriminator:
```suggestion
  condition?: { when: 'always' } | { when: 'overflow', element?: Opt<HtmlElement | VueInstance> }
```

For me both options are good.

Reply:
Yeah, either would be an improvement... but I'm starting to think the complexity of this feature is exceeding its convenience. I think I will reduce the API to an `enabled` prop and let the user call an `isOverflowing` utility.

Reply:
LGTM